reframe/frontend/cli.py

```diff
@@ -429,12 +433,17 @@ def main():
             exec_policy.sched_nodelist = options.nodelist
             exec_policy.sched_exclude_nodelist = options.exclude_nodes
             exec_policy.sched_options = options.job_options
-            runner = Runner(exec_policy, printer)
+            max_retries = options.max_retries
+            runner = Runner(exec_policy, printer, max_retries)
             try:
                 runner.runall(checks_matched, system)
             finally:
-                # always print a report
+                if runner.stats.num_failures(run=0):
```

Comment:
I thought that it would be clearer this way that the retry report is only printed if there were any retries (= if any failures occurred in run 0). Do you still want to remove the if statement?

Reply:
You can leave it as it is and put a comment saying sth like "print a retry report if we had failures in the first run".

Reply:
Done.